tasklib/lazy.py

```diff
@@ -165,12 +165,24 @@ def __le__(self, other):
     def __ge__(self, other):
         return self.issuperset(other)
 
+    def __lt__(self, other):
+        return self.isstrictsubset(other)
+
+    def __gt__(self, other):
+        return self.isstrictsuperset(other)
+
     def issubset(self, other):
         return all([task in other for task in self])
 
     def issuperset(self, other):
         return all([task in self for task in other])
 
+    def isstrictsubset(self, other):
+        return len(set(t['uuid'] for t in other) - self._uuids) > 0
```

Comment:
I think this is actually not a correct definition of a strict subset criterium. Consider the following two sets:
```python
>>> A = {1,2,3}
>>> B = {1,2,4}
>>> B-A
{4}
>>> len(B-A) > 0
True
```

however, the sets are actually incomparable. Same problem holds for the implementation of `isstrictsuperset` below.

For a fix, I'd suggest using `<` and `>` operators for the underlying sets which we're already constructing, like this:
```python
return self._uuids < set(t['uuid'] for t in other)
```

Alternatively, the already existing implementations of `<=` and `>=` could be reused, i.e. the `<` operator could be defined as:
```python
return not self >= other
```

Reply:
Yes, you are absolutely right. I was thinking to fast there 😄
But your second proposed variant is also not working:

```python
>>> A = {1,2,3}
>>> B = {1,2,4}
>>> all([i in A for i in B])  # current implementation of issuperset(self, other)
False
>>> not all([i in A for i in B])
True
```

So I would go with your first suggestion and just use the comparison operator of the `set` class.
In terms of readability it would be nice to also use the `set` operators for `__le__` and `__ge__`. What do you think?

Reply:
>  But your second proposed variant is also not working

Ah yes, that was a brain fart on my end. These UUID sets are not well-ordered, in other words, it does not hold that `A<=B` implies `B<A`.

> So I would go with your first suggestion and just use the comparison operator of the set class.

Sounds good.

>  In terms of readability it would be nice to also use the set operators for __le__ and __ge__. What do you think?

I think this is good refactoring. Existing tests should also help us ensure we don't introduce bugs doing this.

> Edit: Should we also remove the is(strict){subset,superset} methods and implement the functionality directly inside __le__, __ge__, __lt__, __gt__? They do not bring any functionality in that case.

We can't remove existing API, meaning `issubset` and `issuperset` need to stay around, but we don't need to introduce `isstrictsubset` and `isstrictsuperset` (these two methods don't exist in the native `set` object's API either).